nilearn/plotting/surface/_utils.py

```diff
@@ -86,42 +94,111 @@ def _check_bg_map(bg_map, hemi):
     return bg_map
 
 
-def _get_hemi(mesh, hemi):
-    """Check that a given hemisphere exists in a PolyMesh and return the
-    corresponding mesh. If "both" is requested, combine the left and right
+def _get_hemi(surf_mesh, hemi):
+    """Check that a given hemisphere exists in a
+    :obj:`~nilearn.surface.PolyMesh` and return the corresponding
+    ``surf_mesh``. If "both" is requested, combine the left and right
     hemispheres.
+
+    Parameters
+    ----------
+    surf_mesh: :obj:`~nilearn.surface.PolyMesh`
+        The surface mesh object containing the left and/or right hemisphere
+        meshes.
+    hemi: {'left', 'right', 'both'}
+
+    Returns
+    -------
+    surf_mesh : :obj:`numpy.ndarray`,  :obj:`~nilearn.surface.InMemoryMesh`
+        Surface mesh corresponding to the specified ``hemi``.
+
+        - If ``hemi='left'`` or ``hemi='right'``, returns
+          :obj:`numpy.ndarray`.
+        - If ``hemi='both'``, returns :obj:`~nilearn.surface.InMemoryMesh`
     """
+    if not isinstance(surf_mesh, PolyMesh):
+        raise ValueError("mesh should be of type PolyMesh.")
+
     if hemi == "both":
-        return combine_hemispheres_meshes(mesh)
-    elif hemi in mesh.parts:
-        return mesh.parts[hemi]
+        return combine_hemispheres_meshes(surf_mesh)
+    elif hemi in ["left", "right"]:
+        if hemi in surf_mesh.parts:
+            return surf_mesh.parts[hemi]
+        else:
+            raise ValueError(
```

Comment:
that's a legit coverage complaint: see the updloaded HTML coverage report

https://github.com/nilearn/nilearn/actions/runs/14991516206/artifacts/3112733623